### PR TITLE
Implement an adapter from unmanaged to managed client connections

### PR DESCRIPTION
**Pull request by carterkozak**
Draft based on discussion in https://github.com/apache/httpcomponents-core/pull/206#issuecomment-668594741
If this seems helpful I will file a ticket and write tests.

### CODE REVIEW

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/HttpClientConnectionFactoryManagedAdapter.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    @Override
+    public String toString() {
+        return "HttpClientConnectionFactoryManagedAdapter{" + delegate + '}';
+    }
+
+    private static final class ManagedHttpClientConnectionAdapter implements ManagedHttpClientConnection {
+
+        private final HttpClientConnection delegate;
+        private final AtomicBoolean closed = new AtomicBoolean();
```
Comment (carterkozak) [inline httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/HttpClientConnectionFactoryManagedAdapter.java:70]: I'm not sure close tracking is necessary here, but I borrowed it from the DefaultManagedHttpClientConnection implementation.

**Comment by ok2c** [comment]
@carterkozak Is this PR still relevant? Could you please re-base it off the latest master?

**Comment by carterkozak** [comment]
@ok2c I'm not sure this is a helpful utility to provide since we define `DefaultManagedHttpClientConnection` to adapt the httpcomponents-core factory into a managed connection factory, but it was useful to validate new changes from core without a direct dependency from httpcomponents-client onto the snapshot of core. If there are other custom non-managed connection factories, this could be helpful.

I suppose I am leaning towards closing this out, but if you think it would be helpful I'd be more than happy to give it a jira issue and merge it. What do you think?

**Comment by ok2c** [comment]
@carterkozak Let's just close it. We can always re-visit it at a later point.

**Comment by carterkozak** [comment]
Sounds good to me, closed.

**Pull request closed without merging**